backend/database.js

```diff
@@ -52,57 +52,91 @@ export default ((sbp('sbp/selectors/register', {
     }
     // Number of entries pushed.
     let counter = 0
-    let { hash: currentHash, ...serverMeta } = (
-      await sbp('chelonia/db/getEntryMeta', contractID, height)
-    ) || {}
+    let currentHeight = height
```

Comment:
Just above this `o4-mini-high` points out that `process.env.MAX_EVENTS_BATCH_SIZE` isn't coerced to a number:

5) You do not coerce `process.env.MAX_EVENTS_BATCH_SIZE` to a number. As written:

```js
const limit = Math.min(
  requestedLimit ?? Number.POSITIVE_INFINITY,
  process.env.MAX_EVENTS_BATCH_SIZE ?? 500
)
```

If someone exports `MAX_EVENTS_BATCH_SIZE='1000'` your Math.min will see the string `"1000"`.  Always do:

```diff
-  process.env.MAX_EVENTS_BATCH_SIZE ?? 500
+  Number(process.env.MAX_EVENTS_BATCH_SIZE) || 500
```

Reply:
True, but then `Math.min` will coerce it into a number automatically. I'm not sure what the advantage of this is.